vendor/k8s.io/kubernetes/openshift-kube-apiserver/openshiftkubeapiserver/patch.go

```diff
@@ -87,10 +65,10 @@ func NewOpenShiftKubeAPIServerConfigPatch(delegateAPIServer genericapiserver.Del
 
 		// ADMISSION
 		clusterQuotaMappingController := newClusterQuotaMappingController(kubeAPIServerInformers.KubernetesInformers.Core().V1().Namespaces(), kubeAPIServerInformers.OpenshiftQuotaInformers.Quota().V1().ClusterResourceQuotas())
-		patchContext.postStartHooks["quota.openshift.io-clusterquotamapping"] = func(context genericapiserver.PostStartHookContext) error {
+		genericConfig.AddPostStartHookOrDie("quota.openshift.io-clusterquotamapping", func(context genericapiserver.PostStartHookContext) error {
```

Comment:
This causes double-registration as `genericConfig` is later used to contruct `openshiftNonAPIServer`, which is passed to the `StartingDelegate` global variable from `patch_openshift.go` during `buildGenericConfig`.

The delegate variable is then used in `createAPIExtensionsServer()` during its config completion alongside config derived from the `genericConfig` above, so it dies as it tries to merge post-start hooks of these two.

Even if I move this to just `enablement.AddPostStartHookOrDie()`, the API servers initialization fails later in `CreateKubeAPIServer()`with registration of `"openshift.io-kubernetes-informers-synched"`, but would perhaps also fail with the `quota` as it's going to use more or less the same delegate and config for its config completion.

I am not sure how to get out of this double-registration hell.